### ISSUE TITLE
"Invalid URI" being returned on specific URLS

### ISSUE THREAD

**Issue by quantum-x** [Codeception/Codeception#4847]
#### What are you trying to achieve?
Running CodeCeption across CakePHP 2.9.
CakePHP generates URLs / URIs for its pagination in the following format/s:
 - `/units/view/1/sort:id/direction:asc/page:2`
 - `/units/index/page:3/sort:id/direction:asc`

#### What do you get instead?
These URLs/URIs do not parse with Codeception `2.4.0`
> Provide console output if related. Use `-vvv` mode for more details.

```bash
 I click "a[rel="next"]"
  [Request Headers] []
  [Page] /units/view/1/page:2
  [Response] 200
  [Request Cookies] {"CAKEPHP":"qlefnv35v4sh8ojda5a6lfsse0","LANG":"en_US"}
  [Response Headers] {"Date":["Thu, 08 Feb 2018 00:49:01 GMT"],"X-Preferred-Language":["en_US"],"Vary":["X-Preferred-Language,X-User-TZ"],"X-User-TZ":["Asia/Hong_Kong"],"Expires":["Mon, 26 Jul 1997 05:00:00 GMT"],"Last-Modified":["Thu, 08 Feb 2018 00:49:01 GMT"],"Cache-Control":["no-store, no-cache, must-revalidate, post-check=0, pre-check=0"],"Content-Type":["text/html; charset=UTF-8"],"Age":["0"],"X-Cache":["uncached"],"X-Cache-Hits":["0"],"Content-Length":["148950"],"Connection":["keep-alive"],"Accept-Ranges":["bytes"]}
 ERROR

  [Db] Disconnected
--------------------------------------------------------------------------------------------------------------------------------


Time: 5.63 seconds, Memory: 24.00MB

There was 1 error:

---------
1) AdminDevicesCest: Check i can see a particular device and all its scans
 Test  tests/rServices/AdminDevicesCest.php:checkUnitScans

  [InvalidArgumentException] Invalid URI /units/view/1/page:2

```
> Provide test source code if related
```php
$I->click('a[rel="next"]');
```
Trying to bypass the issue doesn't seem to help
```php
$I->amOnPage('/units/view/1/page:2');
```
### Details

* Codeception version: 2.4.0
* PHP Version: 5.6.30
* Operating System: Debian
* Installation type: Phar
* List of installed packages (`composer show`)
* Suite configuration: No json config..

```yml
class_name: RServicesTester
modules:
    enabled:
        - \Helper\RServices
        - PhpBrowser:
            url: http://r.services
        - Db:
        - Asserts:

```

**Comment by quantum-x** [comment]
Older versions (< `2.1.0` ) didn't seem to have this problem.
I think this might be guzzle related - will check into it.

**Comment by Naktibalda** [comment]
Please run codeception with `--debug` flag to get a full stacktrace.

**Comment by quantum-x** [comment]
Thanks for the quick reply. 
Seems to be traced to this update:
> https://github.com/Codeception/Codeception/blame/c50789a9a62cc0eefc0252e88a5f04f8c47b55f4/src/Codeception/Util/Uri.php#L22

Consistent with the fact that I updated an (old) `codeception.phar`

It seems like:
```php
<?php
var_dump(parse_url('/units/view/1/page:2'));
```
Results in:
```
bool(false)
```
Unsure what the best method is to proceed..
Do you still need the debug output ?

**Comment by quantum-x** [comment]
Relevant information: 
 - https://github.com/guzzle/psr7/issues/56
 - https://github.com/cakephp/cakephp/pull/9488

**Comment by quantum-x** [comment]
@Naktibalda This is perhaps a bug in PHP's parse_url, getting confused between URL / URI:
```php
<?php
var_dump(parse_url('/units/view/1/page:2'));
var_dump(parse_url('http://www.goo.com/units/view/1/page:2'));
```


```
bool(false)

array(3) {
  ["scheme"]=>
  string(4) "http"
  ["host"]=>
  string(11) "www.goo.com"
  ["path"]=>
  string(20) "/units/view/1/page:2"
}
```

**Comment by Naktibalda** [comment]
Solved by #4853 4 years ago.
